test/slex_lexer_test.exs

```diff
@@ -0,0 +1,81 @@
+defmodule SlexLexerTest do
+  use ExUnit.Case
+
+  def assert_tokens(input, tokens) do
+    {:ok, output, _} = :slex_lexer.string(input)
+    assert output == tokens
+  end
+
+  test "single element" do
+    assert_tokens 'html', [
+      { :word,      1, 'html' }
+    ]
+  end
+
+  test "single element with id" do
+    assert_tokens 'div#id', [
+      { :word,      1, 'div'  },
+      { :id,        1, 'id'   }
+    ]
+  end
+
+  test "single element with class" do
+    assert_tokens 'div.class', [
+      { :word,      1, 'div'   },
+      { :class,     1, 'class' }
+    ]
+  end
+
+  test "single element with multiple classes" do
+    assert_tokens 'div.c1.c-2', [
+      { :word,      1, 'div'   },
+      { :class,     1, 'c1'    },
+      { :class,     1, 'c-2'   }
+    ]
+  end
+
+  test "element with one attribute" do
+    assert_tokens 'meta a="A"', [
+      { :word,      1, 'meta' },
+      { :space,     1, ' '    },
```

Comment:
Why are spaces not skipped?  Is this a side-effect of them not being skipped for indentation? (which semantically matters).

The parser is going to be uglier if it has to deal with spaces from the token stream for anything other than indentation.